ceph/ceph_admin/upgrade.py

```diff
@@ -57,15 +58,36 @@ def start_upgrade(self: OrchProtocol, config: Dict):
 
         return self.shell(args=cmd)
 
-    def upgrade_status(self: OrchProtocol):
+    def upgrade_status(self: OrchProtocol, timeout: int = 600, interval: int = 10):
         """
         Execute the command ceph orch status.
 
         Returns:
             upgrade Status (Dict)
 
         """
-        out, _ = self.shell(args=["ceph", "orch", "upgrade", "status"])
+        # Retry upgrade command for interval time until it pass till timeout
```

Comment:
My suggestion is to keep this outside `upgrade.py`, maybe in any workflow and not to affect any libraries.

Reply:
@sunilkumarn417 this is just retry and should not affect any upgrades. Kindly check latest upgrade pipeline results.